.github/workflows/on-nextjs-release-publish.yml

```diff
@@ -0,0 +1,71 @@
+# A workflow supposed to run when there is a new release of Next.js.
+# This relies on next.js upstream's `repository_dispatch` workflow.
+name: Next.js Release Publish
+
+on:
+  repository_dispatch:
+    # This is the event type defined by next.js upstream's `repository_dispatch` workflow dispatches.
+    types: [nextjs-release-published]
+  workflow_dispatch:
+    inputs:
+      version:
+        description: Next.js version
+        type: string
+      force_post_to_slack:
+        description: Post test results to Slack
+        type: boolean
+
+jobs:
+  # Debug purpose, write down release version.
+  check-release-tag:
+    name: Check latest release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Print release tag
+        run: echo "Found a new release via ${{ github.event_name }} -- ${{ inputs.version || github.event.client_payload.version }}"
+
+  # Trigger actual next.js integration tests.
+  next_js_integration:
+    name: Execute Next.js integration workflow
+    permissions:
+      pull-requests: write
+    uses: ./.github/workflows/nextjs-integration-test.yml
+    with:
+      force_post_to_slack: ${{ inputs.force_post_to_slack || false }}
+      version: ${{ inputs.version || github.event.client_payload.version }}
+      diff_base: "release"
+
+  # Upload test results to branch.
+  upload_test_results:
+    name: Upload test results
+    # Don't upload results when this workflow is run manually via workflow_dispatch
+    if: ${{ github.event_name != 'workflow_dispatch' }}
+    needs: [next_js_integration]
+    uses: ./.github/workflows/upload-nextjs-integration-test-results.yml
```

Comment:
Could this also be a bit more generalized so we can use it in turborepo?

Reply:
Any specific usecase you have in mind? I can try to generalize, but honestly I'd like to switch this to some other storage if possible.